security/pkg/nodeagent/sds/sdsservice_test.go

```diff
@@ -1023,7 +1023,9 @@ func (ms *mockSecretStore) DeleteSecret(conID, resourceName string) {
 }
 
 func (ms *mockSecretStore) ShouldWaitForIngressGatewaySecret(connectionID, resourceName, token string) bool {
-	return false
+	// Return true here for file mounted cert resource name to validate that file mounted certs would
+	// not depend on this result and proceed with generating secret.
+	return strings.HasPrefix(resourceName, "file-root:")
```

Comment:
Is this implemented for Gateway? I thought it was just DestinationRule

Reply:
It was just implemented for DestinationRule and other server certs. But the sds service https://github.com/istio/istio/blob/master/security/pkg/nodeagent/sds/sdsservice.go#L328 here waits for it